File: app/src/main/java/com/amaze/filemanager/ui/activities/superclasses/BasicActivity.java
package com.amaze.filemanager.ui.activities.superclasses;

import androidx.appcompat.app.AppCompatActivity;

import com.amaze.filemanager.ui.colors.ColorPreferenceHelper;
import com.amaze.filemanager.ui.provider.UtilitiesProvider;
import com.amaze.filemanager.ui.theme.AppTheme;
import com.amaze.filemanager.application.AmazeFileManagerApplication;

public class BasicActivity extends AppCompatActivity {

    protected AmazeFileManagerApplication getAppConfig() {
        return (AmazeFileManagerApplication) getApplication();
    }

    public ColorPreferenceHelper getColorPreference() {
        return getAppConfig().getUtilsProvider().getColorPreference();
    }

    public AppTheme getAppTheme() {
        return getAppConfig().getUtilsProvider().getAppTheme();
    }

    public UtilitiesProvider getUtilsProvider() {
        return getAppConfig().getUtilsProvider();
    }
}

File: app/src/main/java/com/amaze/filemanager/ui/activities/superclasses/BasicActivity.kt
package com.amaze.filemanager.ui.activities.superclasses

import androidx.appcompat.app.AppCompatActivity
import com.amaze.filemanager.application.AmazeFileManagerApplication
import com.amaze.filemanager.ui.colors.ColorPreferenceHelper
import com.amaze.filemanager.ui.provider.UtilitiesProvider
import com.amaze.filemanager.ui.theme.AppTheme

open class BasicActivity : AppCompatActivity() {

    private fun getAppConfig() = application as AmazeFileManagerApplication

    fun getColorPreference(): ColorPreferenceHelper = getAppConfig().utilsProvider.colorPreference

    fun getAppTheme(): AppTheme = getAppConfig().utilsProvider.appTheme

    fun getUtilsProvider(): UtilitiesProvider = getAppConfig().utilsProvider
}